packages/medusa/src/services/fulfillment-provider.ts

```diff
@@ -0,0 +1,177 @@
+import { MedusaError } from "medusa-core-utils"
+import BaseFulfillmentService from "medusa-interfaces/dist/fulfillment-service"
+import { EntityManager } from "typeorm"
+import { TransactionBaseService } from "../interfaces"
+import {
+  Cart,
+  Fulfillment,
+  FulfillmentProvider,
+  LineItem,
+  Order,
+  Return,
+  ShippingMethod,
+  ShippingOption,
+} from "../models"
+import { FulfillmentProviderRepository } from "../repositories/fulfillment-provider"
+import { CreateFulfillmentOrder } from "../types/fulfillment"
+import { FulfillmentOptions } from "../types/fulfillment-provider"
+import { MedusaContainer } from "../types/global"
+
+type FulfillmentProviderContainer = MedusaContainer & {
+  fulfillmentProviderRepository: typeof FulfillmentProviderRepository
+  manager: EntityManager
+}
+/**
+ * Helps retrive fulfillment providers
+ */
+class FulfillmentProviderService extends TransactionBaseService<FulfillmentProviderService> {
+  protected manager_: EntityManager
+  protected transactionManager_: EntityManager | undefined
+  protected readonly container_: FulfillmentProviderContainer
+
+  constructor(container: FulfillmentProviderContainer) {
+    super(container)
+
+    const { manager } = container
+
+    this.container_ = container
+    this.manager_ = manager
+  }
+
+  async registerInstalledProviders(providers: string[]): Promise<void> {
+    return await this.atomicPhase_(async (manager) => {
+      const { fulfillmentProviderRepository } = this.container_
```

Comment:
todo: this should be assigned on the correct property of the class and assigned in the constructor